src/gsy_e/models/area/market_rotators.py

```diff
@@ -53,31 +53,26 @@ class ForwardMarketRotatorBase:
     def __init__(self, markets: ForwardMarketBase):
         self.markets = markets
 
-    @staticmethod
-    def _get_last_slot_to_be_deleted(current_time: DateTime) -> DateTime:
-        """
-        Return last time_slot that will be deleted from the ahead market buffer.
-        """
-
     @staticmethod
     def _is_it_time_to_rotate(current_time: DateTime) -> bool:
         """Return True if it is time to rotate markets."""
 
     def rotate(self, current_time: DateTime) -> None:
         """Delete orders in expired day-ahead markets."""
         if self._is_it_time_to_rotate(current_time):
-            self.markets.delete_orders_in_old_future_markets(
-                last_slot_to_be_deleted=self._get_last_slot_to_be_deleted(
-                    current_time))
+            slots_deleted = []
+            for delivery_time, market_slot_info in self.markets.open_market_slot_info.items():
+                if market_slot_info.closing_time <= current_time:
```

Comment:
AFAIU, the delivery timeframe is after the closing time. Shouldn't we wait until the energy is delivered before we delete the orders? Just in case?

Reply:
The orders should be persisted in the database, therefore no need to maintain them in the simulation after the market has been closed. Main reason for that is in order to offload the exchange with orders that will not be useful (it is not possible to match open orders after the market has closed)